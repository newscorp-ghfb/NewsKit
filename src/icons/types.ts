import {ComponentType} from 'react';
import {MQ} from '../utils/style/types';
import {Theme} from '../theme';
import {LogicalProps} from '../utils/logical-properties';

export interface SvgProps extends React.SVGAttributes<SVGElement> {
  title?: string;
}

export interface NewsKitIconProps extends SvgProps {
  overrides?: {
    stylePreset?: MQ<string>;
    size?: MQ<string>;
  } & LogicalProps;
}

export type NewsKitIcon = ComponentType<
  NewsKitIconProps & {theme?: Theme | undefined}
>;
