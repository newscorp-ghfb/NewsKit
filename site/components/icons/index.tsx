export * from './icon-filled-check-circle';
export * from './icon-filled-cross-circle';
export * from './icon-filled-dark';
export * from './icon-filled-null-dot';
export * from './icon-filled-light';
export * from './icon-filled-circle';
export * from './icon-filled-aa';
export * from './icon-filled-aaa';
export * from './icon-filled-npm';
export * from './icon-filled-chrome';
export * from './icon-filled-firefox';
export * from './icon-filled-safari';
export * from './icon-filled-edge';
export * from './icon-filled-timer';
export * from './icon-filled-expand-less';
export * from './icon-filled-expand-more';
export * from './icon-filled-search';
export * from './icon-filled-github';
