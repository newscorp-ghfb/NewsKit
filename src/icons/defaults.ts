export default {
  icons: {
    stylePreset: 'iconDefault',
  },
};
