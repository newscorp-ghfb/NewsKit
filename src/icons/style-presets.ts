import {StylePreset} from '../theme/types';

export default {
  iconTEST: {
    base: {
      iconColor: '{{colors.red050}}',
    },
    hover: {
      iconColor: '{{colors.green050}}',
    },
  },

  iconPositive: {
    base: {
      iconColor: '{{colors.inkPositive}}',
    },
  },
  iconNegative: {
    base: {
      iconColor: '{{colors.inkNegative}}',
    },
  },
  iconSocialTwitter: {
    base: {
      iconColor: '{{colors.socialTwitter}}',
    },
  },
  iconSocialGitHub: {
    base: {
      iconColor: '{{colors.socialGithub}}',
    },
  },
  iconSocialFacebook: {
    base: {
      iconColor: '{{colors.socialFacebook}}',
    },
  },
  iconSocialInstagram: {
    base: {
      iconColor: '{{colors.socialInstagram}}',
    },
  },
  iconSocialYoutube: {
    base: {
      iconColor: '{{colors.socialYoutube}}',
    },
  },
  iconSocialWhatsApp: {
    base: {
      iconColor: '{{colors.socialWhatsapp}}',
    },
  },
  iconSocialReddit: {
    base: {
      iconColor: '{{colors.socialReddit}}',
    },
  },

  iconDefault: {
    base: {
      iconColor: '{{colors.inkBase}}',
    },
  },
  indeterminateProgressIndicatorPrimary: {
    base: {
      iconColor: '{{colors.inkBrand010}}',
    },
  },
  indeterminateProgressIndicatorSecondary: {
    base: {
      iconColor: '{{colors.inkBase}}',
    },
  },
  indeterminateProgressIndicatorNegative: {
    base: {
      iconColor: '{{colors.inkNegative}}',
    },
  },
  indeterminateProgressIndicatorPositive: {
    base: {
      iconColor: '{{colors.inkPositive}}',
    },
  },
  indeterminateProgressIndicatorInverse: {
    base: {
      iconColor: '{{colors.inkInverse}}',
    },
  },
} as Record<string, StylePreset>;
