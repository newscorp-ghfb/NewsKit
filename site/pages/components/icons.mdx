import Layout from '../../components/layout';
import Playground from '../../components/playground';
import {Prop} from '../../components/prop';
import {PropRenderer as Icon} from '../../components/prop-renderer';
import {TextBoxSwatch} from '../../components/theming-values/colors';
import {LegacyBlock} from '../../components/legacy-block';
import {Code} from '../../components/code';
import {CommonLogicalMarginPropsMDX} from '../../components/component-api/common-logical-props';
import {InlineMessage, Link} from 'newskit';

import {IconFilledFacebook} from '../../components/icons/icon-filled-facebook.tsx';
import {IconFilledGitHub} from '../../components/icons/icon-filled-github.tsx';
import {IconFilledTwitter} from '../../components/icons/icon-filled-twitter.tsx';
import {IconFilledWhatsApp} from '../../components/icons/icon-filled-whatsapp.tsx';
import {IconFilledYoutube} from '../../components/icons/icon-filled-youtube.tsx';
import {IconFilledFigma} from '../../components/icons/icon-filled-figma.tsx';

export default Layout;

# Icons

## Overview

Icons are small SVG shapes, ranging from basic UI shapes to brand logos. NewsKit contains a default set but any of these can be overriden by providing a React component in the icons section of your theme.

### Usage

Icons can be used to enhance other elements and content, as well as used inside links or buttons when space is limited.

### Available Icons

| Name               |                                  Preview                                  |
| ------------------ | :-----------------------------------------------------------------------: |
| IconFilledYoutube  | <Icon overrides={{size: 'iconSize050'}} Component={IconFilledYoutube} />  |
| IconFilledGitHub   |  <Icon overrides={{size: 'iconSize050'}} Component={IconFilledGitHub}/>   |
| IconFilledFacebook | <Icon overrides={{size: 'iconSize050'}} Component={IconFilledFacebook} /> |
| IconFilledTwitter  | <Icon overrides={{size: 'iconSize050'}} Component={IconFilledTwitter} />  |
| IconFilledWhatsApp | <Icon overrides={{size: 'iconSize050'}} Component={IconFilledWhatsApp} /> |
| IconFilledFigma    |  <Icon overrides={{size: 'iconSize050'}} Component={IconFilledFigma} />   |

<InlineMessage
  title="Warning"
  overrides={{stylPreset: 'inlineMessageNegative'}}
>
  From version newskit@6 we don't export emotion icons anymore.
  <br /> Please read bellow how to to import and use emotions icons in your project.
</InlineMessage>

## Props

Icons must have a size specified, and some also support an optional colour override.

<LegacyBlock marginTop="sizing040">
  <Prop name="title" type="string">
    This optional prop adds title to the svg image in order to make it
    accessible and visible for screen readers. If your icon is just decorative
    and not part of the content this prop can be ommited, otherwise it should be
    passed.
  </Prop>
  <Prop name="overrides" type="object">
    If provided, overrides the respective presets for the component and provided
    elements.
    <Prop name="stylePreset" type="MQ<string>">
      If provided, this overrides the style preset applied to the svg.
    </Prop>
    <Prop name="size" type="MQ<string>">
      If provided, this overrides the size of the icon.
    </Prop>
    <CommonLogicalMarginPropsMDX nested />
  </Prop>
</LegacyBlock>

## Playground

<Playground
  componentName={['Facebook', 'Twitter', 'WhatsApp', 'Github']}
  component={[
    IconFilledFacebook,
    IconFilledTwitter,
    IconFilledWhatsApp,
    IconFilledGitHub,
  ]}
  knobs={[
    {
      name: 'Size',
      propName: 'overrides',
      options: [
        {
          label: 'iconSize020',
          value: {
            size: 'iconSize020',
          },
        },
        {
          label: 'iconSize030',
          value: {
            size: 'iconSize030',
          },
        },
        {
          label: 'iconSize040',
          value: {
            size: 'iconSize040',
          },
          isDefault: true,
        },
        {
          label: 'iconSize050',
          value: {
            size: 'iconSize050',
          },
        },
      ],
    },
  ]}
/>

## Create NewsKit icons

With the help of the <a href="/components/utils/custom-to-newskit-icon">`customToNewsKitIcon`</a> util function, each SVG icon can be easily turned into a Newskit icon that works well
with theme tokens and <a href="/components/utils/get-from-theme">`getFromTheme`</a> util functions. The only requirement is to wrap the SVG markup within an `Svg` component and pass props to it.

The following example shows how the SVG component should be used:

<Code>
  {`
  <Svg {...props} viewBox="0 0 429 422">
    <g fill="none" fillRule="evenodd" transform="translate(-77.000000, -1738.000000) translate(77.000000, 1738.500000)">
      <circle cx="250.5" cy="242.5" r="178" style="mix-blend-mode:multiply"/>
      <circle cx="195" cy="195" r="195" style="mix-blend-mode:multiply"/>
    </g>
  </Svg>
 `}
</Code>

Use `customToNewsKitIcon` utility function to make the svg icon compatible with NewsKit:

<Code>
  {`
export const IconFilledFacebook = customToNewsKitIcon(
  'IconFilledFacebook',
  props => (
    <Svg {...props} viewBox="0 0 24 24">
      <path
        fillRule="nonzero"
        d="M13.0784931,22.2595724 L13.0784931,13.1365112 L16.2002321,13.1365112 L16.6676019,9.5811424 L13.0784931,9.5811424 L13.0784931,7.31114568 C13.0784931,6.28180493 13.3699223,5.58025387 14.8748197,5.58025387 L16.7941176,5.57938466 L16.7941176,2.39951562 C16.4620261,2.35634473 15.3227937,2.25957241 13.9973813,2.25957241 C11.2301818,2.25957241 9.33569472,3.916388 9.33569472,6.9592112 L9.33569472,9.58123898 L6.20588235,9.58123898 L6.20588235,13.1366078 L9.33559627,13.1366078 L9.33559627,22.2595724 L13.0784931,22.2595724 Z"
      />
    </Svg>
  ),
);
 `}
</Code>

## Using Emotion icons

From version `newskit@6.0.0` we do not export emotion-icons anymore. In order to contunie using these icons you will need to import them directly from `emotion-icons` package and covert them to NewsKit icons using `customToNewsKitIcon` utility function.

You can find the list of available icons in the <Link href="https://emotion-icons.dev/" target="_blank">emotion-icons.dev</Link> page.

<Code>
  {`
    import AccountTree from '@emotion-icons/material/AccountTree';
    const IconFilledAccountTree = customToNewsKitIcon(
      'IconFilledAccountTree',
      AccountTree,
    );
  `}
</Code>

## Rationale

### Accessibility

Where icon colour can be overidden, the icon must have a 3:1 minimum contrast ratio with its background.
