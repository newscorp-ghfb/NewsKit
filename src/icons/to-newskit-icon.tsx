import React from 'react';
import {EmotionIconProps} from '@emotion-icons/emotion-icon';
import {withTheme} from '../theme';
import {NewsKitIconProps, NewsKitIcon, SvgProps} from './types';
import {getSizingCssFromTheme, getStylePreset, styled} from '../utils/style';
import defaults from './defaults';
import stylePresets from './style-presets';
import {withOwnTheme} from '../utils/with-own-theme';
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const renderIconStylePreset = (overridesOnly: boolean) => (props: any) => {
  if (!overridesOnly || (props.overrides && props.overrides.stylePreset)) {
    return getStylePreset('icons', '', {
      isSvg: true,
    })(props);
  }
  return {};
};

export const toNewsKitIcon = (
  PassedIcon:
    | React.ComponentType<EmotionIconProps>
    | React.ComponentType<SvgProps>,
): NewsKitIcon =>
  withOwnTheme(
    withTheme<NewsKitIconProps>(
      React.memo(props => {
        const emotionIconName = PassedIcon.displayName;
        const toStyledIcon = (
          Icon: React.ComponentType<EmotionIconProps>,
        ) => styled(Icon)`
          // If not overridden, render SP CSS here, this allows parent SP to override Icon default.
          ${renderIconStylePreset(false)}

          vertical-align: unset;
          display: inline-block;
          // https://css-tricks.com/the-sass-ampersand/#doubling-up-specificity
          && {
            //we don't want the icon to have a default size hence using non defaulted functions
            ${props.overrides?.size &&
            getSizingCssFromTheme('width', props.overrides.size)}
            ${props.overrides?.size &&
            getSizingCssFromTheme('height', props.overrides.size)}
        // If overridden, render SP CSS here instead - this ensures we override fill color from parent SP.
        ${renderIconStylePreset(true)}
          }
        `;
        const Icon =
          props.theme.icons[`${emotionIconName}`] ||
          toStyledIcon(PassedIcon as React.ComponentType<EmotionIconProps>);
        return <Icon title={props.title} {...props} />;
      }),
    ),
  )({defaults, stylePresets});
