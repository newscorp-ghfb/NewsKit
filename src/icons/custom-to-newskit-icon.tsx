import {withDefaultProps} from '../utils/with-default-props';
import {toNewsKitIcon} from './to-newskit-icon';
import {NewsKitIconProps, SvgProps} from './types';

export const customToNewsKitIcon = (
  displayName: string,
  CustomIcon: React.ComponentType<SvgProps>,
  overrides?: NewsKitIconProps['overrides'],
) => {
  // eslint-disable-next-line no-param-reassign
  CustomIcon.displayName = displayName;

  if (!overrides) {
    return toNewsKitIcon(CustomIcon);
  }

  return withDefaultProps(toNewsKitIcon(CustomIcon), {
    overrides,
  });
};
