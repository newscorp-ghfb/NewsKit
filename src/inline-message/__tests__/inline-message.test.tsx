import React from 'react';
import {InlineMessage} from '..';
import {renderToFragmentWithTheme} from '../../test/test-utils';
import {IconFilledInfo} from '../../icons';
import {compileTheme, createTheme} from '../..';

const message = 'Your message text here';
const title = 'Your title here';
const icon = <IconFilledInfo overrides={{size: 'iconSize020'}} />;

describe('InlineMessage', () => {
  it('renders with default props', () => {
    const props = {
      children: message,
    };
    const fragment = renderToFragmentWithTheme(InlineMessage, props) as any;
    expect(fragment).toMatchSnapshot();
  });
  it('renders with Title', () => {
    const props = {
      title,
      children: message,
    };
    const fragment = renderToFragmentWithTheme(InlineMessage, props) as any;
    expect(fragment).toMatchSnapshot();
  });
  it('renders with Title and Icon', () => {
    const props = {
      title,
      icon,
      children: message,
    };
    const fragment = renderToFragmentWithTheme(InlineMessage, props) as any;
    expect(fragment).toMatchSnapshot();
  });
  it('renders with overrides and all props', () => {
    const myCustomTheme = compileTheme(
      createTheme({
        name: 'inline-message-theme',
        overrides: {
          stylePresets: {
            customInlineMessage: {
              base: {
                backgroundColor: 'lightpink',
                borderWidth: '1px 4px',
                borderColor: 'deeppink',
                borderStyle: 'solid',
                borderRadius: '3px',
                iconColor: 'deeppink',
              },
            },
          },
        },
      }),
    );
    const props = {
      title,
      icon,
      children: message,
      overrides: {
        stylePreset: 'customInlineMessage',
        paddingInline: 'space050',
        paddingBlock: 'space050',
        icon: {
          spaceInline: 'space050',
        },
        content: {
          message: {
            typographyPreset: 'utilityBody020',
            stylePreset: 'inkInverse',
          },
          title: {
            typographyPreset: 'utilityHeading020',
            stylePreset: 'inkInverse',
            spaceStack: 'space050',
          },
        },
      },
    };
    const fragment = renderToFragmentWithTheme(
      InlineMessage,
      props,
      compileTheme(myCustomTheme),
    ) as any;
    expect(fragment).toMatchSnapshot();
  });
  it('renders with logical prop overrides', () => {
    const props = {
      title,
      icon,
      children: message,
      overrides: {
        paddingInline: '30px',
        marginInline: '50px',
      },
    };
    const fragment = renderToFragmentWithTheme(InlineMessage, props) as any;
    expect(fragment).toMatchSnapshot();
  });
});
