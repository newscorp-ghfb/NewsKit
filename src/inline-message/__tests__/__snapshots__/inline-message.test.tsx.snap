// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineMessage renders with Title 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #E6F4F6;
  border-color: #017582;
  border-style: solid;
  border-radius: 4px;
  border-width: 0 0 0 4px;
  color: #FFFFFF;
  width: 100%;
  padding: 12px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  fill: #017582;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.emotion-3 {
  margin: 0;
  font-family: "Poppins",sans-serif;
  font-size: 16px;
  line-height: 18px;
  font-weight: 700;
  letter-spacing: 0;
  padding: 0.5px 0px;
  color: #09111C;
  margin-bottom: 12px;
}

.emotion-3::before {
  content: '';
  margin-bottom: -0.2155em;
  display: block;
}

.emotion-3::after {
  content: '';
  margin-top: -0.2125em;
  display: block;
}

.emotion-3 svg {
  fill: #09111C;
}

.emotion-4 {
  margin: 0;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 400;
  letter-spacing: 0;
  padding: 0.5px 0px;
  color: #2E3F54;
}

.emotion-4::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-4::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-4 svg {
  fill: #2E3F54;
}

<div
    class="emotion-0"
    data-testid="inline-message"
  >
    <div
      class="emotion-1"
    >
      <div
        aria-live="off"
        class="emotion-2"
        role="region"
      >
        <p
          class="emotion-3"
        >
          Your title here
        </p>
        <p
          class="emotion-4"
        >
          Your message text here
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InlineMessage renders with Title and Icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #E6F4F6;
  border-color: #017582;
  border-style: solid;
  border-radius: 4px;
  border-width: 0 0 0 4px;
  color: #FFFFFF;
  width: 100%;
  padding: 12px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  fill: #017582;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-2 {
  margin-right: 12px;
}

.emotion-2 svg {
  display: block;
}

.emotion-3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  fill: #2E3F54;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-3 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-3 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-3.emotion-3 {
  width: 24px;
  height: 24px;
}

.emotion-5 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.emotion-6 {
  margin: 0;
  font-family: "Poppins",sans-serif;
  font-size: 16px;
  line-height: 18px;
  font-weight: 700;
  letter-spacing: 0;
  padding: 0.5px 0px;
  color: #09111C;
  margin-bottom: 12px;
}

.emotion-6::before {
  content: '';
  margin-bottom: -0.2155em;
  display: block;
}

.emotion-6::after {
  content: '';
  margin-top: -0.2125em;
  display: block;
}

.emotion-6 svg {
  fill: #09111C;
}

.emotion-7 {
  margin: 0;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 400;
  letter-spacing: 0;
  padding: 0.5px 0px;
  color: #2E3F54;
}

.emotion-7::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-7::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-7 svg {
  fill: #2E3F54;
}

<div
    class="emotion-0"
    data-testid="inline-message"
  >
    <div
      class="emotion-1"
    >
      <div
        aria-hidden="true"
        class="emotion-2"
      >
        <svg
          aria-hidden="true"
          class="emotion-3 emotion-4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
      </div>
      <div
        aria-live="off"
        class="emotion-5"
        role="region"
      >
        <p
          class="emotion-6"
        >
          Your title here
        </p>
        <p
          class="emotion-7"
        >
          Your message text here
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InlineMessage renders with default props 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #E6F4F6;
  border-color: #017582;
  border-style: solid;
  border-radius: 4px;
  border-width: 0 0 0 4px;
  color: #FFFFFF;
  width: 100%;
  padding: 12px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  fill: #017582;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.emotion-3 {
  margin: 0;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 400;
  letter-spacing: 0;
  padding: 0.5px 0px;
  color: #2E3F54;
}

.emotion-3::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-3::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-3 svg {
  fill: #2E3F54;
}

<div
    class="emotion-0"
    data-testid="inline-message"
  >
    <div
      class="emotion-1"
    >
      <div
        aria-live="off"
        class="emotion-2"
        role="region"
      >
        <p
          class="emotion-3"
        >
          Your message text here
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InlineMessage renders with logical prop overrides 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #E6F4F6;
  border-color: #017582;
  border-style: solid;
  border-radius: 4px;
  border-width: 0 0 0 4px;
  color: #FFFFFF;
  width: 100%;
  padding: 12px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-inline: 50px;
  padding-inline: 30px;
}

.emotion-0 svg {
  fill: #017582;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-2 {
  margin-right: 12px;
}

.emotion-2 svg {
  display: block;
}

.emotion-3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  fill: #2E3F54;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-3 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-3 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-3.emotion-3 {
  width: 24px;
  height: 24px;
}

.emotion-5 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.emotion-6 {
  margin: 0;
  font-family: "Poppins",sans-serif;
  font-size: 16px;
  line-height: 18px;
  font-weight: 700;
  letter-spacing: 0;
  padding: 0.5px 0px;
  color: #09111C;
  margin-bottom: 12px;
}

.emotion-6::before {
  content: '';
  margin-bottom: -0.2155em;
  display: block;
}

.emotion-6::after {
  content: '';
  margin-top: -0.2125em;
  display: block;
}

.emotion-6 svg {
  fill: #09111C;
}

.emotion-7 {
  margin: 0;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 400;
  letter-spacing: 0;
  padding: 0.5px 0px;
  color: #2E3F54;
}

.emotion-7::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-7::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-7 svg {
  fill: #2E3F54;
}

<div
    class="emotion-0"
    data-testid="inline-message"
  >
    <div
      class="emotion-1"
    >
      <div
        aria-hidden="true"
        class="emotion-2"
      >
        <svg
          aria-hidden="true"
          class="emotion-3 emotion-4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
      </div>
      <div
        aria-live="off"
        class="emotion-5"
        role="region"
      >
        <p
          class="emotion-6"
        >
          Your title here
        </p>
        <p
          class="emotion-7"
        >
          Your message text here
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InlineMessage renders with overrides and all props 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: lightpink;
  border-width: 1px 4px;
  border-color: deeppink;
  border-style: solid;
  border-radius: 3px;
  width: 100%;
  padding: 24px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  fill: deeppink;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-2 {
  margin-right: 24px;
}

.emotion-2 svg {
  display: block;
}

.emotion-3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  fill: #2E3F54;
  vertical-align: unset;
  display: inline-block;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-3 {
    transition-property: fill;
    transition-duration: 200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-3 {
    transition-property: fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-3.emotion-3 {
  width: 24px;
  height: 24px;
}

.emotion-5 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.emotion-6 {
  margin: 0;
  font-family: "Poppins",sans-serif;
  font-size: 18px;
  line-height: 20.25px;
  font-weight: 700;
  letter-spacing: 0;
  padding: 0.5px 0px;
  color: #FFFFFF;
  margin-bottom: 24px;
}

.emotion-6::before {
  content: '';
  margin-bottom: -0.2155em;
  display: block;
}

.emotion-6::after {
  content: '';
  margin-top: -0.2125em;
  display: block;
}

.emotion-6 svg {
  fill: #FFFFFF;
}

.emotion-7 {
  margin: 0;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 400;
  letter-spacing: 0;
  padding: 0.5px 0px;
  color: #FFFFFF;
}

.emotion-7::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-7::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-7 svg {
  fill: #FFFFFF;
}

<div
    class="emotion-0"
    data-testid="inline-message"
  >
    <div
      class="emotion-1"
    >
      <div
        aria-hidden="true"
        class="emotion-2"
      >
        <svg
          aria-hidden="true"
          class="emotion-3 emotion-4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
      </div>
      <div
        aria-live="off"
        class="emotion-5"
        role="region"
      >
        <p
          class="emotion-6"
        >
          Your title here
        </p>
        <p
          class="emotion-7"
        >
          Your message text here
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
